sled-agent/src/hardware/illumos/mod.rs

```diff
@@ -40,6 +40,28 @@ enum Error {
     NoDevLinks(PathBuf),
 }
 
+const GIMLET_ROOT_NODE_NAME: &str = "Oxide,Gimlet";
+
+/// Return true if the host system is an Oxide Gimlet.
+#[cfg(not(test))]
+pub fn is_gimlet() -> anyhow::Result<bool> {
+    let mut device_info = DevInfo::new_force_load()?;
+    let mut node_walker = device_info.walk_node();
+    let Some(root) = node_walker.next().transpose()? else {
+        return Err(anyhow::anyhow!("No nodes in device tree"));
```

Comment:
Style nit - anyhow provides [`bail!`](https://docs.rs/anyhow/latest/anyhow/macro.bail.html) for this use:

```suggestion
        anyhow::bail!("No nodes in device tree");
```